### PR TITLE
Add end-to-end test

### PR DESCRIPTION
**Pull request by jameswestman**
Add a test that runs flat-manager and postrges using docker-compose, and then builds, uploads, and downloads a trivial app. The test takes about 10 minutes.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,6 +31,22 @@ jobs:
...
+
+     - name: Start containers
+       working-directory: ./tests
+       run: docker-compose up -d
+
+     - name: Wait for the containers to finish loading
+       run: sleep 10
+
+     - name: Run test
+       run: docker exec flat-manager ./tests/run-test.py
```
Comment (barthalion) [inline .github/workflows/ci.yml:48]:
```suggestion
       run: docker compose exec flat-manager ./tests/run-test.py
```

I noticed container names created by compose are different between releases, so it's safer to interact with them with compose as well.

File: tests/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
+FROM rust:latest
```
[the remark this answers is not in the document]
Comment (jameswestman) [reply tests/Dockerfile]: Pinned to 1.61-bullseye.

[the remark this answers is not in the document]
Comment (jameswestman) [reply tests/Dockerfile]: `bullseye` will cause issues with the storefront-info PR, which requires libostree >= 2021.5. Should I change that PR to use older ostree versions, or use a different rust image?

[the remark this answers is not in the document]
Comment (barthalion) [reply tests/Dockerfile]: I'm fine with anything that works for you, just keep in mind that `:latest` tends to be too volatile for CI usage.

**Pull request merged**